src/content/docs/core-concepts/query-pipeline.mdx

```diff
@@ -0,0 +1,102 @@
+---
+title: Query Pipeline
+description: A step-by-step introduction on how to use swiftide for generating answers with a query pipeline.
+sidebar:
+  order: 0
+---
+
+After indexing data, you can use a query pipeline to answer questions, using the indexed data, together with various
+transformations, to generate the most relevant context for the LLM.
+
+Query pipelines, similar to indexing pipelines, function as parallel streams.
+
+In the pipeline, a `Query<states::Pending>` starts its journey of going through
+a series of transformations, before being used to generate an answer.
+
+By default, a similarity search is used on a single embedding.
+
+import { Steps } from "@astrojs/starlight/components";
+
+### A query pipeline pipeline step-by-step
+
+<Steps>
+
+1. Start with the default query pipeline that does a similarity search with a single embedding:
+
+   ```rust
+   let pipeline = query::Pipeline::default();
+   ```
+
+2. Then generate subquestions, improving the semantic coverage of the query:
+
+   ```rust
+   pipeline.then_transform_query(query_transformers::GenerateSubQuestions::from_client(openai_client.clone()))
+   ```
+
+   This transformer takes the **current** query, generates subquestions and sets that as the new query.
```

Comment:
I think the fact that the "**current**"  query at this point is a string that is passed into the pipeline when it is invoked might not be clear to all readers.

We could either expand the snippet to be more than just the single line, or maybe just a slightly more descriptive explanation.